tests/cases/fourslash/refactorConvertToOptionalChainExpression_NotForUnknown.ts

```diff
@@ -0,0 +1,7 @@
+/// <reference path='fourslash.ts' />
+
+/////*a*/foo && foo.bar;/*b*/
+
+// Do not offer refactor for unknown symbols.

```

Comment:
In my opinion, we should offer the refactor here, at least for JS, non-strict users. Probably just leave a todo for later though.

Reply:
This was the result of an attempt to get around `isMatchingReference(source, target)` returning true when `source` and `unknown` both have unknown symbols. Otherwise `foo && zoo.bar` will offer a refactor.

Is there a good way to check that they otherwise have matching names when they are not defined?

Reply:
You’d have to use syntax here, not symbols.